### ISSUE TITLE
EACL-2017-Noise Mitigation for Neural Entity Typing and Relation Extraction

### ISSUE THREAD

**Issue by BrambleXu** [BrambleXu/knowledge-graph-learning#97]
**一句话总结：**

将NER和RE进行联合训练，减少两种noisy，NER到RE的error，记忆DS的error。
#95 的先前研究。不推荐看

**资源：**

\- [pdf](<https://arxiv.org/abs/1612.07495>)

\- [code](<https://github.com/hayy2017/noise_mitigation>): theano

**关键字：**

- dataset: 这个数据集有问题，test6000+竟然比train5000+还要多。CF-FIGMENT (URL, 2016b), a dataset published by Yaghoobzadeh and
  Schutze (2015). CF-FIGMENT is derived from ¨a version of ClueWeb (URL, 2016c) in which
  Freebase entities are annotated using FACC1(URL, 2016d; Gabrilovich et al., 2013). CFFIGMENT contains 200,000 Freebase entities that were mapped to 102 FIGER types (Ling and Weld, 2012), divided into train (50%), dev (20%) and test (30%); and a set of 4,300,000 sentences (contexts) containing those entities.
- address two different types of noise in information extraction models: noise from distant supervision and noise from pipeline input features.
- For the first noise type, we introduce multi-instance multi-label learning algorithms using neural network models, and apply them to fine-grained entity typing for the first time
- For the second noise type, we propose ways to improve the integration of noisy entity type predictions into relation extraction.

**笔记：**

methods for automatic knowledge base completion (KBC) are beneficial. Two subtasks of
KBC are entity typing (ET) and relation extraction (RE). We address both tasks in this paper.

The main technique is multi-instance(MI) learning (Riedel et al., 2010). It relaxes the distant supervision assumption to the assumption that at least one instance of a bag (collection of all sentences containing the given entity/entity pair) expresses the type/relationship given in the KB. Multi-instance multi-label (MIML) learning is a generalization of MI in which one bag can have several labels (Surdeanu et al., 2012)。针对DS中的noisy，最常见的就是ML方法，MIML是添加了针对多标签的问题。

In this paper, we introduce two multi-label versions of MI. (i) MIML-MAX takes the maximum
instance for each label. (ii) MIML-ATT applies, for each label, selective attention to the instances. We apply MIML-MAX and MIML-ATT to finegrained ET. 

In contrast to RE, the ET task we consider contains a larger set of labels, with a variety of different granularities and hierarchical relationships. We show that MIML-ATT deals well with noise in corpus-level ET and improves or matches the results of a supervised model based on global embeddings of entities. ET中假设已经知道了entity之间的hierachical信息。

The second type of noise we address in this paper influences the integration of ET into RE. It has been shown that adding entity types as features improves RE models (cf. Ling and Weld (2012), Liu et al. (2014)). However, noisy training data and difficulties of classification often cause wrong predictions of ET and, as a result, noisy inputs to RE. To address this, we propose a joint model of ET and RE and compare it with methods that integrate ET results in a strict pipeline. The joint model performs best. Among the pipeline models, we show that using probabilities instead of binary decisions better deals with noise (i.e., possible ET errors). 为了解决NER产生的error传递到RE中，所以通过联合模型的方式来把二者结合起来, in a strict pipeline.

**模型图：**

![image-20190408102801823](https://ws1.sinaimg.cn/large/006tNc79gy1g1uyx375jlj30vp0i576p.jpg)

**结果**：

![image-20190408103009479](https://ws2.sinaimg.cn/large/006tNc79gy1g1uyzbfp3gj30ji0e40vv.jpg)
